spirv_glsl.cpp

```diff
@@ -2802,7 +2802,9 @@ void CompilerGLSL::emit_texture_op(const Instruction &i)
 	auto &imgtype = get<SPIRType>(type.self);
 
 	// Mark that this shader reads from this image
-	imgtype.image.is_read = true;
+	auto *p_var = maybe_get_backing_variable(img);
+	if (p_var)
+		unset_decoration(p_var->self, DecorationNonReadable);
```

Comment:
I don't understand this. DecorationNonReadable is only used for image load/store textures, not normal sampled textures (they are always read-only in SPIR-V no matter what). Also, changing decorations like this is unsafe while emitting code.

If you want to modify the flags of an image load/store texture, please look at OpImageRead/OpImageWrite implementation:

https://github.com/KhronosGroup/SPIRV-Cross/blob/master/spirv_glsl.cpp#L5387
https://github.com/KhronosGroup/SPIRV-Cross/blob/master/spirv_glsl.cpp#L5520